### PR TITLE
luci-app-aria2: Add package

### PR DESCRIPTION
**Pull request by kuoruan**
LuCI Support for Aria2

Signed-off-by: Hsing-Wang Liao <kuoruan@gmail.com>

### CODE REVIEW

File: applications/luci-app-aria2/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (C) 2016 Openwrt.org
+#
+# This is free software, licensed under the Apache License, Version 2.0 .
+#
+
+include $(TOPDIR)/rules.mk
+
+LUCI_TITLE:=LuCI Support for Aria2
+LUCI_DEPENDS:=+aria2 +!PACKAGE_webui-aria2:yaaw
```
Comment (hnyman) [inline applications/luci-app-aria2/Makefile]: Where does webui_aria2 get defined?

Comment (kuoruan) [inline applications/luci-app-aria2/Makefile]: webui_aria2 is a little big, I am wondering whether to add it.

In fact luci-aria2 does not depend on webui-ariaw nor yaaw.

**Comment by hnyman** [comment]
I am not sure if it pays off to add two different webuis for aria2. If everything works with yaaw, it might be better to stick with that. Or does the Luci part work with either?

My earlier question was because there was an unknown config symbol in the Makefile (as you had not yet introduced the PR for webui-aria2).

File: applications/luci-app-aria2/luasrc/controller/aria2.lua [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+function status()
+	local sys  = require "luci.sys"
+	local ipkg = require "luci.model.ipkg"
+	local http = require "luci.http"
+	local uci  = require "luci.model.uci".cursor()
+
+	local status = {
+		running = (sys.call("pidof aria2c > /dev/null") == 0),
+		yaaw = ipkg.installed("yaaw"),
+		webui = ipkg.installed("webui-aria2")
```
Comment (kuoruan) [inline applications/luci-app-aria2/luasrc/controller/aria2.lua]: There, I add a check for whether installed yaaw or webui-aria2.

File: applications/luci-app-aria2/luasrc/view/aria2/web_script.htm [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	function(x, data) {
+		var tb = document.getElementById('aria2_status');
+		if (data && tb) {
+			if (data.running) {
+				var links = '<em><%:The Aria2 service is running.%></em>';
+				if (data.yaaw) {
+					links += '<input class="cbi-button mar-10" type="button" value="<%:Open YAAW%>" onclick="openWebUI(\'yaaw\');" />';
+				}
+				if (data.webui) {
+					links += '<input class="cbi-button mar-10" type="button" value="<%:Open WebUI-Aria2%>" onclick="openWebUI(\'webui-aria2\');" />';
```
Comment (kuoruan) [inline applications/luci-app-aria2/luasrc/view/aria2/web_script.htm]: And there shows the button to yaaw or webui-aria2.
If not installed either of them, it's OK.
So ```+!PACKAGE_webui-aria2:yaaw``` can be removed

**Comment by kuoruan** [comment]
https://github.com/openwrt/packages/pull/2769

**Comment by hnyman** [comment]
This still requires  https://github.com/openwrt/packages/pull/2756 to work, right?
The aria2 package maintainer has not yet reacted to that PR.

**Comment by kuoruan** [comment]
Yes, This package requires https://github.com/openwrt/packages/pull/2756.

**Pull request merged**